### PR TITLE
[chore] release preparation for 0.16.0

### PR DESCRIPTION
**Pull request by mowies**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,58 @@
...
+
+<details>
+<summary>Highlights from the upstream Collector changelog</summary>
+</br>
+
+### 🛑 Breaking changes 🛑
+
+- `service/telemetry`: Change default metrics address to "localhost:8888" instead of ":8888" ([#11251](https://github.com/open-telemetry/opentelemetry-collector/pull/11251))
+  This behavior can be disabled by disabling the feature gate 'telemetry.UseLocalHostAsDefaultMetricsAddress'.
+- `config`: Move component.UseLocalHostAsDefaultHost feature gate to stable. ([#35569](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35569))
```
Comment (odubajDT) [inline CHANGELOG.md:26]: Maybe I would add a note here, that this option cannot be used anymore, since it cannot be disabled

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,58 @@
...
+  ([#23809](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23809))
+- `hostmetricsreceiver`: Add ability to mute all errors (mainly due to access rights) coming from process scraper of the hostmetricsreceiver ([#20435](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20435))
+- `pkg/ottl`: Add InsertXML Converter ([#35436](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35436))
+- `pkg/ottl`: Add GetXML Converter ([#35462](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35462))
+- `pkg/ottl`: Add ToKeyValueString Converter ([#35334](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35334))
+- `pkg/ottl`: Add RemoveXML Converter ([#35301](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35301))
+
+### 🧰 Bug fixes 🧰
+
+- `processorhelper`: Fix data race condition, concurrent writes to the err variable, causes UB (Undefined Behavior) ([#11350](https://github.com/open-telemetry/opentelemetry-collector/pull/11350))
```
Comment (odubajDT) [inline CHANGELOG.md:53]: is processorhelper part of our distro release?

File: CHANGELOG.md (hunk 1 above)
Comment (mowies) [reply CHANGELOG.md]: i clarified it a bit more, should be fixed now

File: CHANGELOG.md (hunk 2 above)
Comment (mowies) [reply CHANGELOG.md:53]: We definitely had changelog messages about it in previous releases. But it is a transitive dependency i believe

**Pull request merged**